### PR TITLE
SO-4809: Make inclusion of non-property type changes optional during ...

### PR DESCRIPTION
**Pull request by AAAlinaaa**
... compare, exclude them from branch compare results, count total
commit details in new compare result content

### CODE REVIEW

File: commons/com.b2international.index/src/com/b2international/index/revision/DefaultRevisionIndex.java [hunk 1]
```diff
@@ -142,27 +142,27 @@ public RevisionCompare compare(final String branch) {
...
 	@Override
 	public RevisionCompare compare(final String baseBranch, final String compareBranch) {
-		return compare(baseBranch, compareBranch, COMPARE_DEFAULT_LIMIT);
+		return compare(baseBranch, compareBranch, COMPARE_DEFAULT_LIMIT, true);
 	}
 	
 	@Override
-	public RevisionCompare compare(final String baseBranch, final String compareBranch, final int limit) {
-		return compare(getBranchRef(baseBranch), getBranchRef(compareBranch), limit);
+	public RevisionCompare compare(final String baseBranch, final String compareBranch, final int limit, boolean preserveComponentChanges) {
```
Comment (cmark) [inline commons/com.b2international.index/src/com/b2international/index/revision/DefaultRevisionIndex.java]: I think a better configuration option would be the opposite, `excludeComponentChanges`. The default functionality is to always include them, so allowing optional exclusion of them is the functionality we need. Please rename.

Comment (cmark) [reply commons/com.b2international.index/src/com/b2international/index/revision/DefaultRevisionIndex.java]: Alternatively, one could define the set of Operation literal to exclude.

File: commons/com.b2international.index/src/com/b2international/index/revision/RevisionCompare.java [hunk 2]
```diff
@@ -138,14 +144,16 @@ public RevisionCompare build() {
 	private final int totalAdded;
 	private final int totalChanged;
 	private final int totalRemoved;
+	private final int total;
```
Comment (cmark) [inline commons/com.b2international.index/src/com/b2international/index/revision/RevisionCompare.java]: What is total here? Total `added` + `changed` + `removed`?

**Review by cmark — CHANGES_REQUESTED** [review]

File: commons/com.b2international.index/src/com/b2international/index/revision/RevisionCompare.java (hunk 2 above)
Comment (AAAlinaaa) [reply commons/com.b2international.index/src/com/b2international/index/revision/RevisionCompare.java]: No, it is the total, possibly more than the sum of the three if it is more than the limit.

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/branch/compare/BranchCompareResult.java [hunk 3]
```diff
@@ -79,7 +79,7 @@ public Builder totalDeleted(int totalDeleted) {
 			this.totalDeleted = totalDeleted;
 			return this;
 		}
-		
+				
 		public BranchCompareResult build() {
```
Comment (cmark) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/branch/compare/BranchCompareResult.java:83]: I think you can safely remove the `build(Set changedContainers)` method and keep the zero-arg build method as the one that actually builds the result.

**Review by cmark — CHANGES_REQUESTED** [review]

**Review by cmark — APPROVED** [review]
LGTM!

**Pull request merged**